### PR TITLE
Deselect mods incompatible with "Autoplay" when entering gameplay via scene library

### LINKED ISSUE #17936: Skin editor scene library uses autoplay despite selected mods being incompatible with autoplay
### Type

Game behaviour

### Bug description

So I just wanted to add a pp counter using the skin editor but I forgot that I had Magnetised on, however the autoplay still played like there wasn't magnetised on since its supposed to be not compatible.
By doing so you can also make autoplay play with other incompatible mods like No Fail or Spun Out

### Screenshots or videos

https://imgur.com/a/cyflxId

### Version

2022.418.0-lazer

### Logs

[database.log](https://github.com/ppy/osu/files/8542842/database.log)
[network.log](https://github.com/ppy/osu/files/8542843/network.log)
[performance.log](https://github.com/ppy/osu/files/8542844/performance.log)
[runtime.log](https://github.com/ppy/osu/files/8542845/runtime.log)
[updater.log](https://github.com/ppy/osu/files/8542846/updater.log)

### PR DESCRIPTION
**Pull request by frenzibyte**
- Closes #17936 

Also includes simplifying and sharing notification dismiss method across `OsuGameTestScene`s, rather than declaring a "click mouse in centre" method in every test scene.

When incompatible mods are deselected, they'll not return back after finishing from gameplay. Not entirely sure how that should be handled, and I also feel it's unnecessary since I don't think that's how we want users to select mods while testing gameplay (going into song select, selecting mods, then launching skin editor and opening gameplay).

### CODE REVIEW

File: osu.Game/Tests/Visual/OsuGameTestScene.cs [hunk 1]
```diff
@@ -106,6 +108,11 @@ protected void PushAndConfirm(Func<Screen> newScreen)
 
         protected void ConfirmAtMainMenu() => AddUntilStep("Wait for main menu", () => Game.ScreenStack.CurrentScreen is MainMenu menu && menu.IsLoaded);
 
+        /// <summary>
+        /// Dismisses any notifications pushed which block from interacting with the game (or block screens from loading, e.g. <see cref="Player"/>).
+        /// </summary>
+        protected void DismissAnyNotifications() => Game.Notifications.State.Value = Visibility.Hidden;
```
Comment (bdach) [inline osu.Game/Tests/Visual/OsuGameTestScene.cs:114]: Aside from being slightly annoyed that borderline unrelated changes are being shipped together with other changes again, I'd like to ask - mostly for clarification purposes - if there was any particular reason to change how this was implemented from the original version?

Comment (frenzibyte) [reply osu.Game/Tests/Visual/OsuGameTestScene.cs:114]: The point of it was dismissing notifications, and I felt off just clicking in game center for such method, so I simplified it instead.

Apologies for the unrelated changes, but wanted to stop from copying that method once again.

Comment (peppy) [reply osu.Game/Tests/Visual/OsuGameTestScene.cs:114]: I think this method is probably more reliable (although it no longer covers the "click to dismiss" flow? not sure).

**Review by peppy — APPROVED** [review]
